File: 150-evaluate-reverse-polish-notation/evaluate-reverse-polish-notation.java
class Solution {
    public int evalRPN(String[] tokens) {
        int n=tokens.length;
        Stack<Integer> st= new Stack<>();
        for(int i=0; i< n ;i++){
            String str= tokens[i];
            if(str.equals("+")){
                int second= st.pop();
                int first= st.pop();
                st.push(first+second);
            }
            else if(str.equals("-")){
                int second= st.pop();
                int first= st.pop();
                st.push(first-second);
            }
             else if(str.equals("/")){
                int second= st.pop();
                int first= st.pop();
                st.push(first/second);
            }
             else if(str.equals("*")){
                int second= st.pop();
                int first= st.pop();
                st.push(first*second);
            }
            else{
                st.push(Integer.valueOf(str));
            }
        }
        return st.pop();
    }
}

File: 150-evaluate-reverse-polish-notation/evaluate-reverse-polish-notation.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        st=[]

        for token in tokens:
            if token  in ('+', '-', '*',  '/'):
                bottom=int(st.pop())
                top=int(st.pop())
                if token=='+':
                    st.append(top+bottom)
                elif token=='-':
                    st.append( top-bottom)
                elif token=='*':
                    st.append(top*bottom)
                elif token=='/':
                    st.append( top/bottom)

            else:
                st.append(token)
        return int(st[-1])
            

